### PR TITLE
chore: generate events docs

### PR DESCRIPTION
**Pull request by kumaraditya303**
Closes https://github.com/microsoft/playwright-python/issues/786

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -14,7 +14,7 @@ warn_redundant_casts = True
 warn_unused_configs = True
 check_untyped_defs = True
 disallow_untyped_defs = True
-[mypy-tests/async.*]
```
Comment (mxschmitt) [inline setup.cfg:17]: why this? The reason for that was that our sync tests are now 100% typed to ensure that our typings are good.

File: scripts/documentation_provider.py [hunk 2]
```diff
@@ -63,11 +64,12 @@ def _patch_case(self) -> None:
                 self._add_link(member["kind"], clazz["name"], member_name, new_name)
 
                 if member["kind"] == "event":
-                    continue
+                    clazz["events"].append(member)
 
-                new_name = to_snake_case(new_name)
-                member["name"] = new_name
-                members[new_name] = member
+                if member["kind"] != "event":
```
Comment (mxschmitt) [inline scripts/documentation_provider.py:69]:
```suggestion
                else:
```

File: scripts/documentation_provider.py [hunk 3]
```diff
@@ -55,6 +55,7 @@ def _patch_case(self) -> None:
                 continue
             members = {}
             self.classes[clazz["name"]] = clazz
+            clazz["events"] = []
```
Comment (mxschmitt) [inline scripts/documentation_provider.py:58]: lets do it like we do for the `members` variable. Initialise it after line 56 and set it in line 97.

File: scripts/documentation_provider.py [hunk 4]
```diff
@@ -386,7 +421,8 @@ def inner_serialize_doc_type(self, type: Any, direction: str) -> str:
                     )
                 )
             return f"{{{', '.join(items)}}}"
-
+        if type_name == "String":
```
Comment (mxschmitt) [inline scripts/documentation_provider.py:424]: lets drop this and add in line 428 "or type_name == "String" or e.g. lower().

**Comment by mxschmitt** [comment]
nice work!
https://user-images.githubusercontent.com/17984549/133244167-4bd7e033-ad28-44c9-bee1-4fa5d6aeaf4a.mov

**Comment by mxschmitt** [comment]
Can we have a fallback to the value which we had before? Otherwise we might break existing users which had the event name as a string. e.g. at the end of the on overloads in the Page.

```

    @typing.overload
    def on(
        self,
        event: str,
        f: typing.Callable[..., typing.Union[typing.Awaitable[None], None]],
    ) -> None:
        return super().on(event=event, f=f)
```

File: setup.cfg (hunk 1 above)
Comment (kumaraditya303) [reply setup.cfg:17]: That's because of event now emit the objects, feel free to fix the tests mypy

**Comment by kumaraditya303** [comment]
> Can we have a fallback to the value which we had before? Otherwise we might break existing users which had the event name as a string. e.g. at the end of the on overloads in the Page.
> 
> ```
> 
>     @typing.overload
>     def on(
>         self,
>         event: str,
>         f: typing.Callable[..., typing.Union[typing.Awaitable[None], None]],
>     ) -> None:
>         return super().on(event=event, f=f)
> ```

This is non-breaking change moreover it is typing only change hence there is *no*  runtime change

File: playwright/async_api/_generated.py [hunk 5]
```diff
@@ -693,6 +694,100 @@ async def handle(route, request):
...
+        event: Literal["socketerror"],
+        f: typing.Callable[["str"], "typing.Union[typing.Awaitable[None], None]"],
+    ) -> None:
+        """
+        Fired when the websocket has an error."""
+
+    def on(
+        self,
+        event: str,
+        f: typing.Callable[..., typing.Union[typing.Awaitable[None], None]],
```
Comment (kumaraditya303) [inline playwright/async_api/_generated.py:740]: See the fallback here

**Review by mxschmitt — APPROVED** [review]

**Pull request merged**